### PR TITLE
Add logstash helm chart

### PR DESCRIPTION
**Pull request by azro352**
Hi

My contribution to the eck project : the logstash helm chart 

Anyone feel free to review and comment, the goal is to put as examples the files of https://github.com/elastic/cloud-on-k8s/tree/main/config/recipes/logstash

In these examples there is not only a logstash, but also some elastic, beats, ... can you confirm me that I only need to put the Logstash part in the corresponding example file for the helm chart ? 

- [source doc](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-logstash.html)
- [source code](https://github.com/elastic/cloud-on-k8s/blob/e94f7e1caa8ba70449c0bb8ef56eaac63a17dd25/pkg/apis/logstash/v1alpha1/logstash_types.go#L25-L96)

Resolves https://github.com/elastic/cloud-on-k8s/issues/7128

Thanks

### CODE REVIEW

**Comment by azro352** [comment]
I've write the chart for each section of `spec`, I've write the values for the tests

Tell me if you want some things more :)

**Comment by azro352** [comment]
Hi @thbkrkr , Do I have something to do for my PR to be reviewed ? Or just you don't have time for now ?

**Comment by thbkrkr** [comment]
Sorry, it's a question of time and priority. We will review it as soon as possible.

File: deploy/eck-stack/charts/eck-logstash/values.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-logstash-configuration.html#k8s-logstash-configuring-logstash
+#
+config: {}
+
+# Set podTemplate to customize the pod used by Logstash
+# ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-customize-pods.html
+#
+podTemplate: {}
+
+monitoring: {}
```
Comment (naemono) [inline deploy/eck-stack/charts/eck-logstash/values.yaml:46]:
```suggestion
# Settings for configuring stack monitoring.
# ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-stack-monitoring.html
#
monitoring: {}
  # metrics:
  #   elasticsearchRefs:
  #   - name: monitoring
  #     namespace: observability 
  # logs:
  #   elasticsearchRefs:
  #   - name: monitoring
  #     namespace: observability
```

Probably want to be consistent with the ES chart in showing how monitoring is configured.

File: deploy/eck-stack/charts/eck-logstash/values.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#
+# fullnameOverride will override both the release name, and the chart name,
+# and will name the Logstash resource exactly as specified.
+#
+# fullnameOverride: "quickstart"
+
+# Version of Logstash.
+#
+version: 8.10.0-SNAPSHOT
+
```
Comment (naemono) [inline deploy/eck-stack/charts/eck-logstash/values.yaml:22]: We probably also want the ability to override the `image`.

File: deploy/eck-stack/charts/eck-logstash/values.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+pipelines: []
+
+# volumeClaimTemplates
+# ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-logstash-configuration.html#k8s-volume-claim-settings
+#
+volumeClaimTemplates: []
+
+elasticsearchRefs: []
+
+services: []
```
Comment (naemono) [inline deploy/eck-stack/charts/eck-logstash/values.yaml:60]: `secureSettings` and `serviceAccountName` are also valid logstash options that we want to include here to allow override.

Comment (naemono) [reply deploy/eck-stack/charts/eck-logstash/values.yaml:60]: `configRef` and `pipelinesRef` are also valid and both will need validation to ensure only one is configured.

**Review by naemono — CHANGES_REQUESTED** [review]
Looks really good initially. Thanks much for the contribution. There are a couple of things I've initially noticed, and I've pushed a commit doing the following
1. Adding example logstash tests (run them with `make helm-test`)
2. Adding logstash to the eck-stack chart values, and Chart.yaml.

File: deploy/eck-stack/charts/eck-logstash/values.yaml (hunk 3 above)
Comment (azro352) [reply deploy/eck-stack/charts/eck-logstash/values.yaml:71]: Where are documented these keys ? I don't see any of them in https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-logstash.html and its children

I would have added them if I had seen them

Comment (naemono) [reply deploy/eck-stack/charts/eck-logstash/values.yaml:71]: @azro352 These are defined on the struct itself: 
https://github.com/elastic/cloud-on-k8s/blob/e94f7e1caa8ba70449c0bb8ef56eaac63a17dd25/pkg/apis/logstash/v1alpha1/logstash_types.go#L25-L96

**Comment by azro352** [comment]
Test passed


```
Ensuring dependencies are updated for eck-logstash chart.
Running 'helm lint' on eck-logstash chart.
==> Linting .

1 chart(s) linted, 0 chart(s) failed

### Chart [ eck-logstash ] .

 PASS  test logstash    templates/tests/logstash_test.yaml

Charts:      1 passed, 1 total
Test Suites: 1 passed, 1 total
Tests:       11 passed, 11 total
Snapshot:    0 passed, 0 total
Time:        4.26977523s
```

**Comment by azro352** [comment]
Is that a feature you want to have github actions that tests for example the helm charts ?

**Comment by naemono** [comment]
> Is that a feature you want to have github actions that tests for example the helm charts ?

No thank you. There are some buildkite steps that do this that I'll trigger soon after the review. Thanks again for the contributions.

**Comment by naemono** [comment]
I've been testing this a bit more, and found these issue:
```diff
diff --git a/deploy/eck-stack/charts/eck-logstash/templates/logstash.yaml b/deploy/eck-stack/charts/eck-logstash/templates/logstash.yaml
index ce720911a..eeb0d6a25 100644
--- a/deploy/eck-stack/charts/eck-logstash/templates/logstash.yaml
+++ b/deploy/eck-stack/charts/eck-logstash/templates/logstash.yaml
@@ -1,5 +1,5 @@
 ---
-apiVersion: logstash.k8s.elastic.co/v1
+apiVersion: logstash.k8s.elastic.co/v1alpha1
 kind: Logstash
 metadata:
   name: {{ include "logstash.fullname" . }}
@@ -44,13 +44,8 @@ spec:
     {{- toYaml . | nindent 4 }}
   {{- end }}

-  pipelines:
-    {{ toYaml .Values.pipelines | nindent 4 }}
-  volumeClaimTemplates:
-    {{ toYaml .Values.volumeClaimTemplates | nindent 4 }}
-  elasticsearchRefs:
-    {{ toYaml .Values.elasticsearchRefs | nindent 4 }}
-  services:
-    {{ toYaml .Values.services | nindent 4 }}
-  secureSettings:
-    {{ toYaml .Values.secureSettings | nindent 4 }}
+  pipelines: {{ toYaml .Values.pipelines | nindent 4 }}
+  volumeClaimTemplates: {{ toYaml .Values.volumeClaimTemplates | nindent 4 }}
+  elasticsearchRefs: {{ toYaml .Values.elasticsearchRefs | nindent 4 }}
+  services: {{ toYaml .Values.services | nindent 4 }}
+  secureSettings: {{ toYaml .Values.secureSettings | nindent 4 }}
```

The moving of the `toYaml` and adding a `-` is cosmetic, but the v1 => v1alpha1 is not.

**Comment by azro352** [comment]
I'm not sure to understand what are the 2 contents being compared ?

**Comment by RoseSecurity** [comment]
@naemono anyway we can have integrated soon? Looking to utilize this in our infrastructure soon

**Comment by naemono** [comment]
@azro352 I'll update the PR. What I was trying to indicate was that Logstash has no "v1" version, it's only in alpha so it's version is "v1alpha1".

@RoseSecurity We're intending to have this merged and contained within our next release, which is in about a month.

**Review by RoseSecurity — APPROVED** [review]

File: deploy/eck-stack/charts/eck-logstash/templates/logstash.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+---
+apiVersion: logstash.k8s.elastic.co/v1
```
[the remark this answers is not in the document]
Comment (naemono) [reply deploy/eck-stack/charts/eck-logstash/templates/logstash.yaml]: > Will this be changed to `v1` at release or maintain the current?
> 
> * [Source](https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-api-logstash-k8s-elastic-co-v1alpha1.html#k8s-api-logstash-k8s-elastic-co-v1alpha1)

We will be maintaining v1alpha1 at this time, as it hasn't graduated and seen enough use to ensure CRD stability.

**Review by pebrc — COMMENTED** [review]
Looks good. I leave the final 👍 to @naemono. The `pipelinesRef` is the only major thing to fix.

File: docs/orchestrating-elastic-stack-applications/stack-helm-chart.asciidoc [hunk 5]
```diff
@@ -60,6 +60,19 @@ helm install eck-stack-with-fleet elastic/eck-stack \
...
 
+[float]
+[id="{p}-install-logstash-elasticsearch-kibana-helm"]
+== Installing Logstash along with Elasticsearch and Kibana using the eck-stack Helm Chart
+
+The following section builds upon the previous sections, and allows installing Logstash along with Elasticsearch and Kibana.
+
+[source,sh,subs="attributes"]
+----
+# Install an eck-managed Elasticsearch, Kibana, Fleet Server, and managed Elastic Agents using custom values.
```
[the remark this answers is not in the document]
Comment (azro352) [reply docs/orchestrating-elastic-stack-applications/stack-helm-chart.asciidoc:71]: @pebrc what do i do with it ? It comes from a commit of @naemono

[the remark this answers is not in the document]
Comment (naemono) [reply docs/orchestrating-elastic-stack-applications/stack-helm-chart.asciidoc:71]: @azro352 understood. I'll handle this change. This is extremely close to being merged

**Review by naemono — APPROVED** [review]
This LGTM. Everything is functional from my testing, and the docs are in place for this feature. Will merge after CI passes.

**Review by RoseSecurity — APPROVED** [review]

**Comment by naemono** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by naemono** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.10|[](https://github.com/elastic/cloud-on-k8s/pull/7242)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**